ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -249,10 +250,9 @@ public void run() {
       try {
         DataNodeId remoteNode = replicasToReplicatePerNode.get(0).getReplicaId().getDataNodeId();
         ReplicaMetadataResponse response =
-            getReplicaMetadataResponse(replicasToReplicatePerNode, connectedChannel, remoteNode, remoteColo,
-                sslEnabled);
+            getReplicaMetadataResponse(replicasToReplicatePerNode, connectedChannel, remoteNode);
         long startTimeInMs = SystemTime.getInstance().milliseconds();
-        needToWaitForReplicaLag = true;
+        waitEnabled = !replicatingFromRemoteColo;
```

Comment:
we also do it in the constructor. But just that waitEnabled state will change and hence have to re-initialize here